rust/rope/src/interval.rs

```diff
@@ -162,123 +136,122 @@ impl Interval {
 
     // could impl Add trait, but that's probably too cute
     pub fn translate(&self, amount: usize) -> Interval {
-        let amount2 = (amount as u64) * 2;
         Interval {
-            start: self.start + amount2,
-            end: self.end + amount2,
+            start: self.start + amount,
+            end: self.end + amount,
         }
     }
 
     // as above for Sub trait
     pub fn translate_neg(&self, amount: usize) -> Interval {
-        let amount2 = (amount as u64) * 2;
         Interval {
-            start: self.start - amount2,
-            end: self.end - amount2,
+            start: self.start - amount,
+            end: self.end - amount,
         }
     }
 
     // insensitive to open or closed ends, just the size of the interior
     pub fn size(&self) -> usize {
-        self.end() - self.start()
+        self.end().saturating_sub(self.start())
```

Comment:
If there's an end >= start invariant, this can (and probably should) just stay `-`.